nselib/http.lua

```diff
@@ -1571,7 +1571,11 @@ function parse_redirect(host, port, path, response)
   if ( not(u.host) ) then
     -- we're dealing with a relative url
     u.host = stdnse.get_hostname(host)
-    u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    if ( u.path:sub(1,1) == "/") then
+      u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
```

Comment:
Good job on catching this bug. As nnposter said, the first statement is repeated. You could write something like the following to eliminate it:
```
if ( u.path:sub(1,1) ~= "/") then -- If we are working with relative with relative paths
      u.path = ((path:sub(1,1) == "/" and "" ) or ("/")) .. path
                .. ((path:sub(#path,#path) == "/" and "" ) or "/" ) .. u.path -- normalize the path
```
However, this will behave incorrectly in certain cases. What if function parameter 'path' is passed by a call like http.get(host, port, "/folder/folder2/test")? So the base path needs to be extracted to be concatenated to the relative path obtained. Also remember that relative paths can be in this "form" or "./form" or "../form" or "../../form", etc...If a relative uri with a relative path is received, it needs to be normalized (see http-sitemap-generator->normalize_path).

For clarity, the function parameter name 'path' could be renamed to 'basepath' to avoid confusions to which path we are referring to inside the function.